### ISSUE TITLE
Ignore messages by type

### ISSUE THREAD

**Issue by wilt00** [thelounge/thelounge#4166]
### Feature Description

Add `/ignoretype` , `/unignoretype`, and `/ignoretypelist` commands to the server, analogous to existing `/ignore` and `/unignore` commands. This would have the effect of ignoring types of message on a per-network basis.

Usage:
```
/ignoretype JOIN
```

Here's my use case: I've been using TF as, among other things, a cache for Twitch chat for about six months, and it's worked beautifully, with one caveat - channels are so choked by JOIN and PART spam that in some cases (mostly for channels that haven't streamed recently) if I scroll back through history, the tab becomes unresponsive before I hit any message that isn't JOIN or PART. Numbers-wise, my `{username}.sqlite3` database is ~800 MB large, over 400MB of which is Twitch JOIN and PART messages. I don't see any value in archiving these messages, and I'd like to drop them permanently on receipt.

I don't think this is a duplicate of #2573 , since that issue deals with arbitrary pattern matching on usernames and message content, or of #2581 , which would be a rework of the existing command.

I'd be happy to contribute a PR myself for this or a similar feature. My first ham-fisted crack at this, before I realized the scale of the problem, was to not SELECT these messages from the database on load for specific channels (branch here: https://github.com/thelounge/thelounge/compare/master...wilt00:hide-join-history) but I don't think that goes far enough.

**Comment by fnutt** [comment]
I don't think this will be implemented since TL implemented condense messages. In the issue/pr it was discussed different settings, and decided to not have a setting for each message type.

Please see https://github.com/thelounge/thelounge/pull/759 for more information regarding this.

I suggest closing this issue.

**Comment by wilt00** [comment]
Yes, condensed messages are enabled. This is not a cosmetic issue - the client is being overloaded by the volume of useless messages it is trying to process. Here is a video recording of one of the channels I'm in:

https://user-images.githubusercontent.com/7883508/109581104-6e230e00-7ac1-11eb-9197-626b54df7b5d.mp4

**Comment by fnutt** [comment]
So this is not an issue about hiding join/quit messages, but about loading actual messages on scroll. If I remember correctly there is already an issue on the tracker regarding this.

**Comment by xnaas** [comment]
I know #828, but not sure if that's relevant to this specifically or not.
